pkg/ingester/flush.go

```diff
@@ -248,15 +248,18 @@ func (i *Ingester) flushUserSeries(userID string, fp model.Fingerprint, immediat
 }
 
 func (i *Ingester) collectChunksToFlush(instance *instance, fp model.Fingerprint, immediate bool) ([]*chunkDesc, labels.Labels, *sync.RWMutex) {
-	stream, ok := instance.streams.LoadByFP(fp)
+	var stream *stream
+	var ok bool
+	stream, ok = instance.streams.LoadByFP(fp)
 
 	if !ok {
 		return nil, nil, nil
 	}
 
-	var result []*chunkDesc
 	stream.chunkMtx.Lock()
 	defer stream.chunkMtx.Unlock()
+
+	var result []*chunkDesc
```

Comment:
Would a `stream.chunkMtx.RLock()` suffice in line 259?

Reply:
There is a write operation in line 268